### PR TITLE
Refactor wrap_preserving_code into helper utilities

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- extract a `LineBuffer` helper and other utilities so `wrap_preserving_code` no longer manages every concern inline
- route the wrapping control flow through the new helpers to clarify punctuation handling, splits, and trailing whitespace logic
- add focused tests for the helper behaviours and token grouping scenarios

## Testing
- make fmt
- make lint
- make test

------
https://chatgpt.com/codex/tasks/task_e_68d502e4898c832288984f8bc44f8d70

## Summary by Sourcery

Refactor wrap_preserving_code by extracting line buffering and token‐grouping logic into dedicated helper functions to simplify wrapping control flow and punctuation/whitespace handling, and add targeted unit tests for each helper.

Enhancements:
- Extract LineBuffer struct and related utilities to manage buffer state and splitting logic in wrap_preserving_code
- Consolidate token span determination, punctuation attachment, overflow handling, and new‐line logic into separate helper functions
- Simplify wrap_preserving_code by routing control flow through the new helpers

Tests:
- Add focused unit tests for determine_token_span, append_group_to_line, split overflow, trailing whitespace, and new line grouping behaviors

### CODE REVIEW

File: src/wrap/tests.rs [hunk 1]
```diff
@@ -6,7 +6,130 @@
...
+#[case("`code`!", "`code`!")]
+#[case("[link](url).", "[link](url).")]
+#[case("plain,", "plain,")]
+fn determine_token_span_groups_related_tokens(#[case] input: &str, #[case] expected_group: &str) {
+    let tokens = segment_inline(input);
+    let (end, width) = determine_token_span(&tokens, 0);
+    let grouped = tokens[..end].join("");
+    assert_eq!(grouped, expected_group);
+    assert_eq!(width, unicode_width::UnicodeWidthStr::width(expected_group));
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/wrap/tests.rs:34]: Has this now been resolved?

File: src/wrap.rs [hunk 2]
```diff
@@ -108,122 +109,218 @@ fn flush_current(lines: &mut Vec<String>, current: &mut String) {
...
+
+    if let Some(last_line) = lines.last_mut()
+        && last_line.trim_end().ends_with('`')
+    {
+        last_line.push_str(token);
+        return true;
+    }
+
+    false
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/wrap.rs]: if-let chain is stable from Rust 1.88.0 when the crate uses the 2024 edition.

As this is an application and not a library, we are targeting the latest stable version of Rust (v1.89.0).

**Comment by leynos** [comment]
@coderabbitai pause
